src/PotentialCalculation/SuccessiveOverRelaxation/GPU_kernel.jl

```diff
@@ -1,77 +1,81 @@
 @inline function sor_kernel(
     potential::AbstractArray{T, 4},
+    imp_scale::AbstractArray{T, 4},
     point_types::AbstractArray{PointType, 4},
     volume_weights::AbstractArray{T, 4},
     q_eff_imp::AbstractArray{T, 4},
     q_eff_fix::AbstractArray{T, 4},
     ϵ_r::AbstractArray{T, 3},
-    geom_weights::NTuple{3, <:AbstractArray{T, 2}},
+    geom_weights::NTuple{3, AbstractArray{T, 2}},
     sor_const::AbstractArray{T, 1},
     update_even_points::Bool,
     depletion_handling_enabled::Bool,
     is_weighting_potential::Bool,
     only2d::Bool,
     ::Type{S},
-    linear_idx
+    gpu_inds::NTuple{3, Int}
 ) where {T, S}
-    eff_size = broadcast(idim -> 2:size(potential, idim)-1, (1, 2, 3))
-    if linear_idx <= prod(length.(eff_size))
-        i1, i2, i3 = Tuple(CartesianIndices(eff_size)[linear_idx])
-        # Comparison to CPU indices: (Cyl / Car)
-        # i3 <-> idx3 / ir / iz
-        # i2 <-> idx2 / iφ / iy
-        # i1 <-> idx1 / iz / ix
-        in3 = i3 - 1
-        in2 = i2 - 1
-        in1 = nidx(i1, update_even_points, iseven(i2 + i3))
-        i1r = get_rbidx_right_neighbour(i1, update_even_points, iseven(i2 + i3))
-        
-        rb_tar_idx, rb_src_idx = update_even_points ? (rb_even::Int, rb_odd::Int) : (rb_odd::Int, rb_even::Int) 
-
-        geom_weights_3 = get_geom_weights_outerloop(geom_weights, in3, S)
-        geom_weights_2 = prepare_weights_in_middleloop(
-            geom_weights, S, i2, in2, 
-            geom_weights_3...,
-            in3 == 1
-        )
-        weights = calculate_sor_weights(
-            in1, 
-            S, 
-            ϵ_r,
-            geom_weights[3],
-            i2, in2, i3, in3,
-            geom_weights_2...
-        )
+    # Comparison to CPU indices: (Cyl / Car)
+    # i3 <-> idx3 / ir / iz
+    # i2 <-> idx2 / iφ / iy
+    # i1 <-> idx1 / iz / ix
```

Comment:
Did you test this on a non-symmetric detector in `Cartesian` coordinates to see if the indices are not mixed up?